src/ServiceControl.Infrastructure/Watchdog.cs

```diff
@@ -87,25 +88,28 @@ public Task Start(Action onFailedOnStartup)
                         //Ignore, no need to log cancellation of delay
                     }
                 }
-                try
-                {
-                    log.Debug($"Stopping watching process {taskName}");
-                    //We don't pass the shutdown token here because it has already been cancelled and we want to ensure we stop the ingestion.
-                    await ensureStopped(CancellationToken.None).ConfigureAwait(false);
-                }
-                catch (Exception e)
-                {
-                    log.Error($"Error while trying to stop {taskName}.", e);
-                    reportFailure(e.Message);
-                }
-            });
+            }, cancellationToken);
+
             return Task.CompletedTask;
         }
 
-        public Task Stop()
+        public async Task Stop(CancellationToken cancellationToken)
         {
-            shutdownTokenSource.Cancel();
-            return watchdog;
+            try
+            {
+                log.Debug($"Stopping watching process {taskName}");
+                await shutdownTokenSource.CancelAsync().ConfigureAwait(false);
```

Comment:
Nitpick: We should need configureawait in this codebase

Reply:
Should or should not?

Reply:
@andreasohlund I've searched for `.ConfigureAwait(` and there are 82 matches